plugins/inputs/logfile/logfile.go

```diff
@@ -149,8 +149,6 @@ func (t *LogFile) FindLogSrc() []logs.LogSrc {
 
 	t.cleanUpStoppedTailerSrc()
 
-	// If a log group has differing retentionInDays values defined in multiple places, stop the agent
-	t.checkForDuplicateRetentionSettings()
```

Comment:
@SaxyPandaBear Since the agent was not intended to be used by directly supplying the toml file, I don't think its a case we should necessarily support. Restarting the agent through an external source seems like something that would be more common and I do agree, without the agent code validation we wouldn't be protected from that. 

Currently when the agent gets restarted from an external source, wouldn't we be in danger from other invalid inputs?

Reply:
> Currently when the agent gets restarted from an external source, wouldn't we be in danger from other invalid inputs as well?

Yes, and that's why knowing this, I don't want to add something that contributes to this issue.

Reply:
I would like to avoid duplicate code.
The typical and only supported usage for CloudWatch Agent is to give a JSON configuration and then run the translator to convert to TOML.
I do not think we should worry about customers modifying TOML directly.

Checking configuration validity during translation seems preferable to waiting until the Agent program begins.
It happens earlier and the error output can be returned directly to stdout/stderr instead of users having to check log files.

Reply:
I'm on board if we're all in agreement that if a customer messes up their TOML file, that's completely on them. Is this documented in some way?

Reply:
I don't think we explicitly document that we convert Json to Toml (we do print this in the logs) and I think it makes sense. This use case is for customers who have understood how the CWagent flow is working internally but we shouldn't have to add extra support for use cases that we didn't design and test the agent for.

Reply:
> output can be returned directly to stdout/stderr instead of users having to check log files

The output still shows up in log files, assuming that people use the agent through the ctl script. But either way, good enough for me.